pkg/synchromanager/clustersynchro_manager.go

```diff
@@ -282,8 +292,18 @@ func (manager *Manager) reconcileCluster(cluster *clusterv1alpha2.PediaCluster)
 	}
 
 	manager.UpdateClusterSynchroCondition(cluster.Name, clusterv1alpha2.RunningConditionReason, "", metav1.ConditionTrue)
+	syncResources := cluster.Spec.SyncResources
+	if templateName := cluster.Spec.ClusterSyncResources; templateName != "" {
+		template, err := manager.clusterSyncResourcesLister.Get(templateName)
+		if err != nil && !apierrors.IsNotFound(err) {
+			klog.ErrorS(err, "Failed to get cluster sync template", "cluster", cluster.Name)
+		}
+		if template != nil {
+			syncResources = mergeResources(syncResources, template.Spec.SyncResources)
```

Comment:
Instead of using map to de-duplicate, just append the two slices together and clustersynchro will take care of the de-duplication.

```go
syncResources = append(syncResources, template.Spec.SyncResources)
```